File: arena-backend/src/entities/group-participant.entity.ts
import {
  Column,
  Entity,
  JoinColumn,
  OneToOne,
  PrimaryColumn,
} from "typeorm";
import { ParticipantEntity } from "./participant.entity";

export type GroupParticipantRole = "owner" | "member";

@Entity({ name: "group_participants" })
export class GroupParticipantEntity {
  @PrimaryColumn({ type: "text" })
  channelId: string;

  @PrimaryColumn({ type: "text" })
  userId: string;

  @OneToOne(() => ParticipantEntity)
  @JoinColumn([
    { name: "channelId", referencedColumnName: "channelId" },
    { name: "userId", referencedColumnName: "userId" },
  ])
  participant: ParticipantEntity;

  @Column({ type: "varchar", length: 20 })
  role: GroupParticipantRole;

  @Column({ type: "varchar", length: 32, nullable: true })
  nickname: string | null;
}

File: legacy/backend/src/main/java/app/sandori/arena/api/domain/channel/GroupParticipantEntity.java
package app.sandori.arena.api.domain.channel;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "group_participants")
@IdClass(ParticipantId.class)
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class GroupParticipantEntity {

    @Id
    private String channelId;

    @Id
    private String userId;

    @Column(nullable = false, length = 20)
    private String role;

    @Column(length = 32)
    private String nickname;

    public GroupParticipantEntity(String channelId, String userId, String role, String nickname) {
        this.channelId = channelId;
        this.userId = userId;
        this.role = role;
        this.nickname = nickname;
    }
}
